File: Greedy/javascript/1931.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');

let arr = [];
for (let i = 1 ; i<input.length-1 ; i++){
    arr.push(input[i].split(' ').map(Number));
}
arr.sort((a,b) => {
    if (a[1] === b[1]) {
        return a[0]-b[0];
    }
    return a[1]-b[1];
})
let endTime = 0
let result = 0;
for (let i = 0 ; i<arr.length ; i++){
  if (endTime <= arr[i][0]) {
    result++;
    endTime = arr[i][1];
  }
}
console.log(result)

File: Greedy/python/1931.py
n = int(input())
arr = sorted([list(map(int,input().split())) for _ in range(n)], key = lambda x : (x[1],x[0]))
end_time = 0
cnt = 0
for table in arr:
  if table[0] >= end_time:
    end_time = table[1]
    cnt+=1
print(cnt)